### PR TITLE
Correct mappings of `Cooling with room sensor` for F series

### PR DESCRIPTION
**Pull request by tomrennen**
Reverts yozik04/nibe#81, (partially) `fixes` https://github.com/home-assistant/core/issues/89027

Incorrect assumptions were made, parsing of my heatpump in HA is now broken:
```
2023-03-08 11:41:58.572 WARNING (MainThread) [nibe.nibe.connection.nibegw] Ignoring coil 47340 value 0a000000 - failed to decode
2023-03-08 11:41:58.575 DEBUG (MainThread) [homeassistant.components.nibe_heatpump] Some coils failed to update, and may be unsupported: Failed decoding response for cooling-with-room-sensor-47340: Failed to decode cooling-with-room-sensor-47340 coil from raw: 0a000000, exception: Raw value 10 is out of range for coil cooling-with-room-sensor-47340
```

@elupus [the logic](https://github.com/home-assistant/core/blob/dev/homeassistant/components/nibe_heatpump/climate.py#L142) mentioned in the original PR mentions "ON" state, so presumably this value can be mapped to on/off, but how? Maybe the values are `10=Off, 20=On`? How could I confirm this? Can I take any steps to help you with parsing it correctly?

### CODE REVIEW

**Comment by yozik04** [comment]
> Maybe the values are 10=Off, 20=On?

Maybe. The only way to check is to toggle in the menu and see what is received when you read the value with the library.

**Comment by elupus** [comment]
We have a problem too that has surface that one broken read breaks many sensors. Not had the opportunity to look at it yet.

**Comment by yozik04** [comment]
@elupus You mean when read of one fails then others are also failing?

**Comment by elupus** [comment]
Yes seems so. It should return a group exception after finishing the generator data. Maybe i missed something in the update with new exception types. Also noticed there was some exception group fix that came in shortly after the bump we made in HA, so maybe its related.

**Comment by yozik04** [comment]
@elupus You have a unit test for that and it covers that case. Strange. Exception group fix is for Python 3.11.

**Comment by tomrennen** [comment]
I will try to manually check for the values whenever this sensor comes back in HA, it's not that critical for me and easier than writing a python script to check it manually, and create a new PR accordingly. For now this should be merged to fix the mistake I've made.

**Comment by tomrennen** [comment]
Alright, I got that to work.

I've noticed a thing or two:

My `SMO40` allows for selection of `Cooling with room sensor` not just `ON/OFF` but `OFF, BT50, RMU-BT50`. I'd consider both values of `BT50` and `RMU-BT50` to correspond to "On". The mapping is as follows:

```
OFF=000000
BT50=02000000
RMU-BT50=0a000000
```

**Comment by yozik04** [comment]
Mapping this registry to just ON and OFF is wrong. Then you will get a switch entity in Home Assistant. But this should be a selection entity. Just lookup how it is done for others.

**Comment by elupus** [comment]
Just add mappings for those values and it will end up a select entity.

**Comment by tomrennen** [comment]
Alright, I will update this to reflect a selector then. This will break [this logic](https://github.com/home-assistant/core/blob/f4b8598979219038f98593faf0a2f09b7aad8889/homeassistant/components/nibe_heatpump/climate.py#L142), will create a PR for both repos when I have the time.

**Comment by yozik04** [comment]
Is it only for SMO40?

**Comment by tomrennen** [comment]
I wouldn't be able to tell you, the SMO40 is all I own.

**Comment by elupus** [comment]
Please apply this change to all F series pumps. They are all equally broken her.

File: nibe/data/extensions.json [hunk 1]
```diff
@@ -29,6 +29,15 @@
         "mappings": {
           "0": "Off"
         }
+      },
+      "47340": {
+        "factor": 1,
+        "max": 18.0,
+        "mappings": {
+          "0": "OFF",
```
Comment (elupus) [inline nibe/data/extensions.json:37]: Could you also add "1": "ON" here. Could be that some pumps do return that value. If we don't have it listed those pumps will break.

Comment (tomrennen) [reply nibe/data/extensions.json:37]: Good point.

**Review by elupus — APPROVED** [review]

**Pull request merged**